### PR TITLE
Merge master HEAD into openj9-staging

### PR DESCRIPTION
**Pull request by JasonFengJ9**
Resolved conflicts:
```
	src/java.base/share/classes/java/lang/Thread.java
```

W/ this PR and supporting loom changes, `openj9-staging` builds and `-version` output:
```
openjdk version "19-internal" 2022-09-20
OpenJDK Runtime Environment (build 19-internal-adhoc.jasonfeng.testbuild-0515)
Eclipse OpenJ9 VM (build loom_support3-me-9389fc4725, JRE 19 Mac OS X amd64-64-Bit Compressed References 20220515_000000 (JIT enabled, AOT enabled)
OpenJ9   - 9389fc4725
OMR      - c945d253d
JCL      - 26d001ca158 based on jdk-19+21)
```

Note: these changes require OpenJ9 loom support to be promoted into `openj9` branch though it fixes the merge conflict and allows `Merge-to-Staging-JDKNext` job continue content update.

In addition, this enables loom development at this extension repo, and move away from feature extension repo exclusively for project loom.

fyi @tajila @fengxue-IS 

Signed-off-by: Jason Feng <fengj@ca.ibm.com>

### CODE REVIEW

File: src/java.base/share/classes/java/lang/Thread.java [hunk 1]
```diff
@@ -1985,53 +2970,194 @@ public UncaughtExceptionHandler getUncaughtExceptionHandler() {
...
+
+    private void stop0(Object o) {
+        synchronized (interruptLock) {
+            if (o != null) {
+                if (started) {
+                    /* [PR CMVC 179978] Java7:JCK:java_lang.Thread fails in all platform */
+                    /*
+                        * If the thread has not yet been simply store the fact that stop has been called,
+                        * the JVM uses this to determine if stop has been called before start.
+                        */
```
Comment (keithc-ca) [inline src/java.base/share/classes/java/lang/Thread.java:3103]: Please correct the indentation of these lines.

**Review by keithc-ca — CHANGES_REQUESTED** [review]
This is incomplete. Attempting to `make java.base-java` there are over 100 errors. Here is a sample:
```
/home/keithc/space/openj9/jdk19/src/java.base/share/classes/java/lang/ThreadBuilders.java:27: error: cannot find symbol
import java.lang.Thread.Builder;
                       ^
  symbol:   class Builder
  location: class Thread
```
and others apparently referring to classes nested within `Thread.Builder`
```
/home/keithc/space/openj9/jdk19/build/normal/support/j9jcl/java.base/share/classes/java/lang/ref/Reference.java:56: error: <anonymous java.lang.ref.Reference$1> is not abstract and does not override abstract method <T>newNativeReferenceQueue() in JavaLangRefAccess
                SharedSecrets.setJavaLangRefAccess(new JavaLangRefAccess() {
                                                                           ^
  where T is a type-variable:
    T extends Object declared in method <T>newNativeReferenceQueue()
```

```
/home/keithc/space/openj9/jdk19/src/java.base/share/classes/java/net/Socket.java:584: error: reference not found
     *        {@linkplain Thread#isVirtual() virtual thread} is establishing a
                          ^
```
/home/keithc/space/openj9/jdk19/src/java.base/share/classes/java/lang/ThreadLocal.java:175: error: cannot find symbol
        return get(Thread.currentCarrierThread());
                         ^
  symbol:   method currentCarrierThread()
  location: class Thread
```
```
/home/keithc/space/openj9/jdk19/src/java.base/share/classes/jdk/internal/vm/Continuation.java:473: error: cannot find symbol
                + " [" + currentCarrierThread().threadId() + "]";
                                               ^
  symbol:   method threadId()
  location: class Thread
```

**Comment by JasonFengJ9** [comment]
> This is incomplete. Attempting to make java.base-java there are over 100 errors. 

Those errors when built w/ default configuration were due to [JEP 425 Implementation of Virtual Threads (Preview)](https://github.com/ibmruntimes/openj9-openjdk-jdk/commit/e8736df6a0c65356c3836c2d58487e648e972fc4) which can only be built w/ `--enable-openjdk-thread-support` (also `--enable-loom-support`) along w/ OpenJ9 loom changes.
`openj9-staging` branch won't be able to build until loom support code ready.
On the other hand, this PR allows `Merge-to-Staging-JDKNext` job continue content update, in addition, this enables loom development at this extension repo instead of feature extension repo exclusively for project loom.

File: src/java.base/share/classes/java/lang/Thread.java [hunk 2]
```diff
@@ -360,54 +589,466 @@ public static void sleep(long millis) throws InterruptedException {
...
             }
         }
+        initialize(false, g, parent, acc, characteristics);
+
+        if (attached && VM.initLevel() < 1) {
+            this.tid = 1;  // primordial thread
+        } else {
+            this.tid = ThreadIdentifiers.next();
+        }
+
```
[the remark this answers is not in the document]
Comment (JasonFengJ9) [reply src/java.base/share/classes/java/lang/Thread.java]: Removed.

File: src/java.base/share/classes/java/lang/Thread.java (hunk 1 above)
Comment (JasonFengJ9) [reply src/java.base/share/classes/java/lang/Thread.java]: Updated.

**Comment by tajila** [comment]
> hose errors when built w/ default configuration were due to [JEP 425 Implementation of Virtual Threads (Preview)](https://github.com/ibmruntimes/openj9-openjdk-jdk/commit/e8736df6a0c65356c3836c2d58487e648e972fc4) which can only be built w/ --enable-openjdk-thread-support (also --enable-loom-support) along w/ OpenJ9 loom changes.

Note, we won't be using `--enable-loom-support` anymore we'll just use JDK19 JPP/CPP flags instead

**Comment by keithc-ca** [comment]
I can't find `--enable-loom-support` anywhere: was that a plan that never happened?

**Comment by keithc-ca** [comment]
Even with `--enable-openjdk-thread-support`, I see 20 errors compiling `java.base`. Please share an openj9 pull request URL that would allow at least that, or correct those issues in this pull request.

**Comment by JasonFengJ9** [comment]
This PR builds w/ `openj9` - https://github.com/eclipse-openj9/openj9/compare/master...JasonFengJ9:loomV3?expand=1 and `GensrcJ9JCL.gmk`, `OpenJ9.gmk`, `custom-hook.m4`, and `custom-spec.gmk.in` from https://github.com/JasonFengJ9/openj9-openjdk-jdk/tree/mergetmp-loom, which are still in progress as per https://github.com/ibmruntimes/openj9-openjdk-jdk/pull/441#issuecomment-1128003180.

**Comment by keithc-ca** [comment]
Tobi's comment https://github.com/ibmruntimes/openj9-openjdk-jdk/pull/441#issuecomment-1128003180 suggests we don't want the changes in https://github.com/JasonFengJ9/openj9-openjdk-jdk/tree/mergetmp-loom, or said another way,
```
LOOM_SUPPORT == (OPENJDK_THREAD_SUPPORT && (JAVA_SPEC_VERSION >= 19))
```

**Comment by JasonFengJ9** [comment]
The change for `GensrcJ9JCL.gmk`, `OpenJ9.gmk`, `custom-hook.m4`, and `custom-spec.gmk.in` will be in a separated PR, which will be coordinated w/ openj9 loom changes.

**Comment by tajila** [comment]
@JasonFengJ9 We originally planned to add loom JPP/CPP flags when we thought we would support the EA repo before it was merged into the openjdk head stream. Since that is not the case, Ive told Jack to just use the jdk19 JPP/CPP flags. So this `LOOM_SUPPORT == (OPENJDK_THREAD_SUPPORT && (JAVA_SPEC_VERSION >= 19))` is the current status

**Comment by JasonFengJ9** [comment]
> So this LOOM_SUPPORT == (OPENJDK_THREAD_SUPPORT && (JAVA_SPEC_VERSION >= 19)) is the current status

That makes sense, the `j.l.Thread` in this PR is include w/ such `LOOM_SUPPORT` in place.
Those two personal branches were based on Jack earlier version, and for sanity build against this PR as per request of https://github.com/ibmruntimes/openj9-openjdk-jdk/pull/441#issuecomment-1128038267.

**Review by keithc-ca — APPROVED** [review]
With eclipse-openj9/openj9#15073 (modified a little) I was able to build locally with these changes. We'll need 15073 to be merged before things can promote to the openj9 branch, but this is a necessary step on the way.

**Pull request merged**